### DISCUSSION TITLE
Documentation

### DISCUSSION THREAD

**Discussion by Vimm2** [EmulatorJS/EmulatorJS#262] [Ideas]
I've known about emulatorjs.com for a while but dismissed it because it's too mysterious.  There's no explanation of what it is or how it works and the authors are anonymous.  The only content is a morsel of documentation to install it on your site.  That simplicity's great but that's a big ask from an opaque source.  I like to understand what I put on my site so injecting anonymous code from a strange site is a non-starter.  After all, what's to stop them from, say, showing porn ads on my site?

This project gives me control of the content but it's no less mysterious.  Pick any console for instance.  What emulator is it?  What version?  How do I update it?  Why must I dig for such basic information?  The absence of documentation makes me uneasy.

Another example, how do I customize the default options such as video settings or controller mappings?   Maybe it's not possible, or is it?  Again, no documentation.

Keep up the good work!

**Comment by ethanaobrien** [comment]
Thanks for the voice. I’ve been meaning to do this for a while... haven’t gotten around to it. If there’s any more things I need to document, then let me know!

I’m going to make this an issue

**Reply by KingIzzymon** [reply]
Hey @ethanaobrien , I've been using this for a while and I'm fairly decent at writing documentation. Is there anyway I can help flush out the information on your new page https://emulatorjs.org/?

**Reply by ethanaobrien** [reply]
Sure! You can create a branch of this repo - https://github.com/EmulatorJS/emu-site - and can edit from there. It would be helpful to document the features of the emulator itself, the in-game features like dragging and dropping a save file will load it. Please note you will need nodejs to be able to write the docs, as they use vuepress. Thank you for your help! Ask me if you have any questions!

**Reply by KingIzzymon** [reply]
I appreciate the info! Will start working on it tonight. Especially how save files can be uploaded to the server and downloaded to other browsers. Excellent features!

**Reply by ethanaobrien** [reply]
Actually, all developer options are already documented (see https://emulatorjs.org/docs/Options.html ). I meant more features the user can use, like in the newest update I added the ability to drag and drop any save file onto the emulator as it is running and it will be loaded

**Reply by KingIzzymon** [reply]
Hey boss, I'm just about done populating your website with info, but your build instructions for EmulatorJS are missing alot. All you have is how to compile your own version of retroarch-web (I actually have an all-in-one bash script if you want that).

I'd like to add the build instructions before I make the official pull request. Whenever you get the chance. Thx

**Reply by KingIzzymon** [reply]
Also what are your intensions with the "Code Edit" feature on the website? I'm not entirely sure what its supposed to do. Like... portableize/offline-erate a rom+emulator?

**Reply by ethanaobrien** [reply]
> I actually have an all-in-one bash script if you want that

I would love to add that 

> Also what are your intensions with the "Code Edit" feature on the website

The code edit feature is more for those who have to clue how to program and want to make a website. I've had several issues opened about how to use the emulator, so I wrote the code editor to do that for those people

**Reply by KingIzzymon** [reply]
Understood, I'll get that info added in as well.

Going back to the build info question, is there no build instructions for EmulatorJS because it's already ready for production? Just needs to be webhosted via something like NGINX server?

The last part confusing me is your affiliation with linuxserver's EmulatorJS. They say that you are just cleaning up the obfuscated cores but then you claim to be the "official" EmulatorJS project.

Are they handing over the reins to you to eventually be the primary project? Or are you both using the same name to confuse people...?

**Reply by ethanaobrien** [reply]
>Going back to the build info question, is there no build instructions for EmulatorJS because it's already ready for production

Correct, the only compiling needed is minifying for production, which that should already be written

> The last part confusing me is your affiliation with linuxserver's EmulatorJS. They say that you are just cleaning up the obfuscated cores but then you claim to be the "official" EmulatorJS project.

The frontend was obfuscated, but it's been de-obfuscated for some time now. That description in their readme is very outdated. The backend (the cores themselves) has been 100% redone since they put that in their readme

>Are they handing over the reins to you to eventually be the primary project? Or are you both using the same name to confuse people...?

This project started about a year earlier then linuxservers emulatorjs. The fact it has the same name is really confusing, I opened an issue about it being confusing but he kind of dismissed my renaming idea. See this issue - https://github.com/linuxserver/emulatorjs/issues/24

**Reply by KingIzzymon** [reply]
Okay... I just went through and read that whole discussion. So they based their project off of your old https://www.emulatorjs.com/ website and then loaded it with goodies like their [HyperSpin](https://hyperspin-fe.com/)-like frontend, the ROM scraper and save data/file manager.

Just to clarify, your project is ONLY upload a ROM each time to play and download its save state? I spent most of yesterday & last night writing documentation based off of linuxserver's emulatorjs docker container...

**Reply by ethanaobrien** [reply]
>So they based their project off of your old https://www.emulatorjs.com/ website and then loaded it with goodies like their [HyperSpin](https://hyperspin-fe.com/)-like frontend, the ROM scraper and save data/file manager.

My project was initially based upon emulatorjs.com. My initial goal was to remove ads and remove outside dependencies. His project uses some of the old parts of my project.

>Just to clarify, your project is ONLY upload a ROM each time to play and download its save state?

It is not, If you look in the readme at the supported devices it has the instructions for how to configure each system like [nes for example.
](https://github.com/EmulatorJS/EmulatorJS/blob/main/docs/Systems/NES-Famicom.md) The goal of this project is to give developers an easy way to embed an emulator onto their website. The only real place you can upload a rom is in the demo page, which is mainly to show developers the features, performance, ect, even though I know mostly kids just use it for just playing games.

>I spent most of yesterday & last night writing documentation based off of linuxserver's emulatorjs docker container...

Yeah, I'm not affiliated with them, sorry about that.

**Reply by adeniyisikirat3** [reply]
> > So they based their project off of your old https://www.emulatorjs.com/ website and then loaded it with goodies like their [HyperSpin](https://hyperspin-fe.com/)-like frontend, the ROM scraper and save data/file manager.
> 
> My project was initially based upon emulatorjs.com. My initial goal was to remove ads and remove outside dependencies. His project uses some of the old parts of my project.
> 
> > Just to clarify, your project is ONLY upload a ROM each time to play and download its save state?
> 
> It is not, If you look in the readme at the supported devices it has the instructions for how to configure each system like [nes for example. ](https://github.com/EmulatorJS/EmulatorJS/blob/main/docs/Systems/NES-Famicom.md) The goal of this project is to give developers an easy way to embed an emulator onto their website. The only real place you can upload a rom is in the demo page, which is mainly to show developers the features, performance, ect, even though I know mostly kids just use it for just playing games.
> 
> > I spent most of yesterday & last night writing documentation based off of linuxserver's emulatorjs docker container...
> 
> Yeah, I'm not affiliated with them, sorry about that.

[ Yup, you're the boss 👍 , & You did good. I dislike those ads aswell ]

**Reply by hamidshabeer1-ship-it** [reply]
Hello

**Comment by hamidshabeer1-ship-it** [comment]
App download

**Reply by joethun** [reply]
Vhat
